src/lib.rs

```diff
@@ -194,6 +196,41 @@ impl ManagerClient {
     }
 }
 
+#[pyclass(get_all, set_all)]
+struct QuorumResult {
+    quorum_id: i64,
+    replica_rank: i64,
+    replica_world_size: i64,
+    recover_manager_address: String,
+    recover_rank: Option<i64>,
+    recovering_ranks: Vec<i64>,
+    store_address: String,
+    max_step: i64,
+    max_rank: Option<i64>,
+    max_world_size: i64,
+    heal: bool,
+}
+
+#[pymethods]
+impl QuorumResult {
+    #[new]
+    fn new() -> Self {
+        Self {
+            quorum_id: 0,
+            replica_rank: 0,
+            replica_world_size: 1,
+            recover_manager_address: "".to_string(),
+            recover_rank: None,
```

Comment:
nit: many rename `recover_rank` to `recovery_source_rank`? Or add comment that clarifies this represents the rank from which the current replica will recover

Reply:
renamed to `recover_src_rank` and `recover_dst_ranks`